### PR TITLE
spec: document create/delete/list interaction

### LINKED ISSUE #110: spec: clarify expectations related to possibly failed CreateVolume calls and volume deletion
continuation of discussion here: https://github.com/container-storage-interface/spec/pull/88#issuecomment-330918121

* there's an implied relationship between LIST_VOLUMES and CREATE_DELETE capabilities for some cases
* we're expecting that a DC operator may be involved with cleanup for some cases

The spec should talk about these things so that it's clear to both COs and plugins what the specific expectations are.

### PR DESCRIPTION
**Pull request by jdef**
Fixes #110

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -997,6 +997,19 @@ message ControllerServiceCapability {
...
+#### RPC Interactions
+
+##### `CreateVolume`, `DeleteVolume`, `ListVolumes`
+
+It is worth noting that the plugin-generated `volume_id` is a REQUIRED field for the `DeleteVolume` RPC, as opposed to the CO-generated volume `name` that is REQUIRED for the `CreateVolume` RPC: these fields MAY NOT contain the same value.
+If a `CreateVolume` operation times out, leaving the CO without an ID with which to reference a volume, and the CO *also* decides that it no longer needs/wants the volume in question then the CO MAY choose one of the following paths:
+
+1. Replay the `CreateVolume` RPC that timed out; upon success execute `DeleteVolume` using the known volume ID (from the response to `CreateVolume`).
+2. Execute the `ListVolumes` RPC to possibly obtain a volume ID that may be used to execute a `DeleteVolume` RPC; upon success execute `DeleteVolume`.
+3. Do not execute any additional RPCs and instead log an event/message in order to facilitate garbage collection of unused resources at some later point in time.
```
Comment (jieyu) [inline spec.md]: Maybe remove this? I think the garbage collector itself will either use 1 or 2 above to cleanup the unneeded volume?

Comment (saad-ali) [reply spec.md]: Option 3 could be leak and expect user to cleanup.

**Comment by saad-ali** [comment]
LGTM once nit is addressed.

**Comment by jdef** [comment]
@saad-ali @jieyu I rewrote option 3, PTAL

**Comment by jieyu** [comment]
LGTM!

**Pull request merged**